File: src/Java/Visualization/ProteinRibbons.java
import vtk.vtkActor;
import vtk.vtkNamedColors;
import vtk.vtkNativeLibrary;
import vtk.vtkRenderWindow;
import vtk.vtkRenderWindowInteractor;
import vtk.vtkRenderer;
import vtk.vtkProteinRibbonFilter;
import vtk.vtkPDBReader;
import vtk.vtkInteractorStyleSwitch;
import vtk.vtkPolyDataMapper;

public class ProteinRibbons
{
  // -----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------
  

  public static void main(String args[]) 
  {
	  
    //parse command line arguments
    if (args.length != 1) 
    {
      System.err.println("Usage: java -classpath ... Filename(.pdb) e.g 2j6g.pdb");
      return;
    }
    String Filename = args[0];
  
    vtkNamedColors colors = new vtkNamedColors();

    //Renderer Background Color
    double Bgcolor[] = new double[4];

    colors.GetColor("Silver", Bgcolor);
    
    // read protein from pdb
    vtkPDBReader reader = new vtkPDBReader();
    reader.SetFileName(Filename);

    // setup ribbon filter
    vtkProteinRibbonFilter ribbonFilter = new vtkProteinRibbonFilter();
    ribbonFilter.SetInputConnection(reader.GetOutputPort());

    // setup poly data mapper
    vtkPolyDataMapper polyDataMapper = new vtkPolyDataMapper();
    polyDataMapper.SetInputConnection(ribbonFilter.GetOutputPort());

    // setup actor
    vtkActor actor = new vtkActor();
    actor.SetMapper(polyDataMapper);
    actor.GetProperty().SetDiffuse(.7);
    actor.GetProperty().SetSpecular(.5);
    actor.GetProperty().SetSpecularPower(80.0);

    
    // Create the renderer, render window and interactor.
    vtkRenderer ren = new vtkRenderer();
    vtkRenderWindow renWin = new vtkRenderWindow();
    renWin.AddRenderer(ren);
    vtkRenderWindowInteractor iren = new vtkRenderWindowInteractor();
    iren.SetRenderWindow(renWin);
    
    vtkInteractorStyleSwitch style = new vtkInteractorStyleSwitch();
    style.SetCurrentStyleToTrackballCamera();
    
    // Visualise
    ren.AddActor(actor);
    ren.ResetCameraClippingRange();
    ren.SetBackground(Bgcolor);
    
    renWin.SetSize(300, 300);
    renWin.SetMultiSamples(0);
    renWin.Render();

    iren.Initialize();
    iren.Start();
  }
}

File: src/Cxx/Visualization/ProteinRibbons.cxx
#include <vtkProteinRibbonFilter.h>

#include <vtkActor.h>
#include <vtkCamera.h>
#include <vtkPolyDataMapper.h>
#include <vtkNew.h>
#include <vtkProperty.h>
#include <vtkRenderWindowInteractor.h>
#include <vtkRenderWindow.h>
#include <vtkRenderer.h>
#include <vtkPDBReader.h>
#include <vtkInteractorStyleSwitch.h>

#include <vtkNamedColors.h>

int main(int argc, char *argv[])
{
  if (argc < 2)
  {
    std::cout << "Usage: " << argv[0] << " file.pdb" << std::endl;
    return EXIT_FAILURE;                                                                                
  }
  std::string fileName(argv[1]);

  // read protein from pdb
  vtkNew<vtkPDBReader> reader;
  reader->SetFileName(fileName.c_str());

  // setup ribbon filter
  vtkNew<vtkProteinRibbonFilter> ribbonFilter;
  ribbonFilter->SetInputConnection(reader->GetOutputPort());

  // setup poly data mapper
  vtkNew<vtkPolyDataMapper> polyDataMapper;
  polyDataMapper->SetInputConnection(ribbonFilter->GetOutputPort());

  // setup actor
  vtkNew<vtkActor> actor;
  actor->SetMapper(polyDataMapper);
  actor->GetProperty()->SetDiffuse(.7);
  actor->GetProperty()->SetSpecular(.5);
  actor->GetProperty()->SetSpecularPower(80.0);

  // setup render window
  vtkNew<vtkRenderer> renderer;
  vtkNew<vtkRenderWindow> renderWindow;
  renderWindow->AddRenderer(renderer);
  vtkNew<vtkRenderWindowInteractor> interactor;
  interactor->SetRenderWindow(renderWindow);
  vtkInteractorStyleSwitch* style =
    dynamic_cast<vtkInteractorStyleSwitch*>(interactor->GetInteractorStyle());
  style->SetCurrentStyleToTrackballCamera();

  vtkNew<vtkNamedColors> colors;
  renderer->AddActor(actor);
  renderer->SetBackground(colors->GetColor3d("Silver").GetData());
  renderWindow->SetSize(640, 480);
  
  renderer->ResetCamera();
  renderer->GetActiveCamera()->Zoom(1.5);
  renderer->ResetCameraClippingRange();
  renderWindow->Render();

  // Finally render the scene
  renderWindow->SetMultiSamples(0);
  interactor->Initialize();
  interactor->Start();

  return EXIT_SUCCESS;
}
